hydromt/workflows/grid.py

```diff
@@ -21,6 +26,133 @@
 ]
 
 
+def grid_from_region(
+    kind: str,
+    geom: gpd.GeoDataFrame,
+    *,
+    logger: logging.Logger = logger,
+    data_catalog: Optional[DataCatalog] = None,
+    res: Optional[float] = None,
+    crs: Optional[int] = None,
+    rotated: bool = False,
+    hydrography_fn: Optional[str] = None,
+    add_mask: bool = True,
+    align: bool = True,
+    dec_origin: int = 0,
+    dec_rotation: int = 3,
+) -> xr.Dataset:
+    # Derive xcoords, ycoords and geom for the different kind options
+    if kind in ["bbox", "geom"]:
+        if not isinstance(res, (int, float)):
+            raise ValueError("res argument required for kind 'bbox', 'geom'")
+        # Generate grid based on res for region bbox
+        # TODO add warning on res value if crs is projected or not?
```

Comment:
Should we do anything with this TODO? Is there an issue?

Reply:
This was already in the original code